roles/openstack/tasks/server.yaml

```diff
@@ -0,0 +1,141 @@
+---
+# (c) Copyright IBM Corporation
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Extract OS info from image name of server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_os_data: >-
+      {{
+        {
+          'type': server_config.image.split('-')[0] | lower,
+          'version': server_config.image.split('-')[1].split('.')[0:2] | join('.')
+        }
+      }}
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Extract cloud-init yaml for server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_cloud_init: "{{ server_config.cloud_init | from_yaml }}"
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Create empty list for network nics to reset network nics for every server
+  ansible.builtin.set_fact:
+    _neerali_openstack_network_nics: []
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Generate nics payload for server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_network_nics: >-
+      {{
+        _neerali_openstack_network_nics +
+        [
+          {
+            'net-name' : network_name
+          }
+        ]
+      }}
+  loop: "{{ server_config.networks }}"
+  loop_control:
+    loop_var: network_name
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+    - server_config.networks is defined
+
+- name: Create cloud server instances for {{ server_config.name }}
+  vars:
+    _neerali_openstack_server_prefix: "{{ range(1000,9999) | random }}"
+  openstack.cloud.server:
+    cloud: "{{ clouds }}"
+    state: present
+    name: "{{ _neerali_openstack_server_prefix }}-{{ server_config.name }}"
+    image: "{{ server_config.image }}"
+    flavor: "{{ server_config.flavor }}"
+    nics: "{{ _neerali_openstack_network_nics }}"
+    volumes: "{{ server_config.volumes | default([]) }}"
+    timeout: "{{ api_timeout }}"
+    userdata: "{{ server_config.cloud_init }}"
+  async: "{{ api_timeout }}"
+  poll: 0
+  register: _neerali_openstack_async_servers
+  with_sequence: count="{{ server_config.count }}"
```

Comment:
Instead of `with_sequence` can we switch to 
```
loop: "{{ range(server_config.count) | list }}"
```

[Reference](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#with-sequence)

Reply:
done